File: src/test/java/uk/gov/justice/digital/hmpps/oauth2server/security/AuthIpSecurityTest.java
package uk.gov.justice.digital.hmpps.oauth2server.security;

import org.junit.jupiter.api.Test;
import org.springframework.mock.web.MockHttpServletRequest;

import java.util.Set;

import static org.assertj.core.api.Assertions.assertThat;

class AuthIpSecurityTest {

    private AuthIpSecurity testClass;

    @Test
    void testStandardV4IP() {

        final var request = new MockHttpServletRequest();
        request.setRemoteAddr("127.0.0.1");

        testClass = new AuthIpSecurity(Set.of("0.0.0.0/0"));

        final var check = testClass.check(request);

        assertThat(check).isTrue();
    }

    @Test
    void testRemoteAddressWithPort() {

        final var request = new MockHttpServletRequest();
        request.setRemoteAddr("82.34.12.11:13321");

        testClass = new AuthIpSecurity(Set.of("0.0.0.0/0"));

        final var check = testClass.check(request);

        assertThat(check).isTrue();
    }

    @Test
    void testRemoteAddressWithPortNoInWhitelist() {

        final var request = new MockHttpServletRequest();
        request.setRemoteAddr("82.34.12.11:13321");

        testClass = new AuthIpSecurity(Set.of("82.34.12.10/32", "82.34.12.12/32"));

        final var check = testClass.check(request);

        assertThat(check).isFalse();
    }

    @Test
    void testIpV6Address() {

        final var request = new MockHttpServletRequest();
        request.setRemoteAddr("0:0:0:0:0:0:0:1");

        testClass = new AuthIpSecurity(Set.of("0:0:0:0:0:0:0:1", "127.0.0.1/32"));

        final var check = testClass.check(request);

        assertThat(check).isTrue();
    }
}

File: src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/AuthIpSecurityTest.kt
package uk.gov.justice.digital.hmpps.oauth2server.security

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.mock.web.MockHttpServletRequest

class AuthIpSecurityTest {
  @Test
  fun testStandardV4IP() {
    val request = MockHttpServletRequest()
    request.remoteAddr = "127.0.0.1"
    val testClass = AuthIpSecurity(setOf("0.0.0.0/0"))
    val check = testClass.check(request)
    assertThat(check).isTrue()
  }

  @Test
  fun testRemoteAddressWithPort() {
    val request = MockHttpServletRequest()
    request.remoteAddr = "82.34.12.11:13321"
    val testClass = AuthIpSecurity(setOf("0.0.0.0/0"))
    val check = testClass.check(request)
    assertThat(check).isTrue()
  }

  @Test
  fun testRemoteAddressWithPortNoInAllowlist() {
    val request = MockHttpServletRequest()
    request.remoteAddr = "82.34.12.11:13321"
    val testClass = AuthIpSecurity(setOf("82.34.12.10/32", "82.34.12.12/32"))
    val check = testClass.check(request)
    assertThat(check).isFalse()
  }

  @Test
  fun testIpV6Address() {
    val request = MockHttpServletRequest()
    request.remoteAddr = "0:0:0:0:0:0:0:1"
    val testClass = AuthIpSecurity(setOf("0:0:0:0:0:0:0:1", "127.0.0.1/32"))
    val check = testClass.check(request)
    assertThat(check).isTrue()
  }
}
